src/main/java/com/google/sps/servlets/InputServlet.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.image.ImageSelection;
+import com.google.sps.data.Input;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.image.ImageSelection;
+import com.google.sps.data.TextAnalyser;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/text-input")
+public class InputServlet extends HttpServlet {
+
+  private Map<String, String> requests;
```

Comment:
Unfortunately, I do not think this will work reliably the way you expect. App Engine will scale your application to _zero_ instances (i.e. shut down everything that runs your server code) when the web app does not receive any requests. Even when your App Engine app has running instances, you cannot get a guarantee that the same request will hit the same App Engine instance.

To avoid these problems, you need to write your code in a way such that functionality doesn't rely on server state stored in e.g. fields in the Java servlets.

Reply:
fixed